checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -144,6 +144,12 @@ private void checkCreatesMustCallForOverrides(
     }
   }
 
+  @Override
+  protected boolean shouldPerformContractInference() {
+    // TODO: should be "false whenever running MustCallInferenceLogic", probably
+    return false;
```

Comment:
This might no longer be necessary, because IIRC the problem this was "solving" was inference of too many unverifiable `@EnsuresCalledMethod` annotations, which should have been solved recently by #5825.

Reply:
@Nargeshdb I suggest you merge that into this branch, remove this logic, and then check to see if there are still unverifiable `@EnsuresCalledMethods` annotations in the benchmarks. If not, then removing this code is safe, which will make this PR simpler. If so, that will give us another test case that will help us improve the logic from #5825.